core/spec.md

```diff
@@ -4035,9 +4039,34 @@ and the following Resource level attributes:
   - `none` - No compatibility checking is performed.
 
 - Constraints:
-  - If present, it MUST be a case sensitive value from the model defined
+  - If present, it MUST be a case-sensitive value from the model-defined
     enumeration range.
   - When not specified, the default value MUST be `none`.
+  - The value MAY change only when the new `compatibility` value can be
+    enforced across all existing Versions. If that's not the case, the
+    server MUST generate an error.
+
+#### `enforcecompatibility`
+- Name: `enforcecompatibility`
+- Type: Boolean
+- Description: Indicates a request to the server to enforce the `compatibility`
+  value defined by the owning Resource.
+  When set to `true`:
+- 1. The server MUST reject the request if the server cannot enforce the
+     compatibility for this type of Resource.
```

Comment:
Perhaps just say: ...cannot enforce the desired compatibility for the data to be validated for the Resource's Versions.

1) I wasn't sure what "type of Resource" meant since we never really talk about "contenttype" in all of this.
2) technically, someone could define a compat mode that checks more than the doc, it could look at the xreg data too. In fact, it should to some extent to check the "contenttype", but in general if hasdoc=false then "compat" for that kind of resource might involve checking the xReg metadata across versions.

So I think the more generic statement covers all future complex cases too